MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -2014,19 +1996,29 @@ private string GetFixedSizeScalingFilter(Video3DFormat? threedFormat, int reques
             }
 
             // If we're hardware VAAPI decoding and software encoding, download frames from the decoder first
-            else if (string.Equals(options.HardwareAccelerationType, "vaapi", StringComparison.OrdinalIgnoreCase)
+            else if (IsVaapiSupported(state) && string.Equals(options.HardwareAccelerationType, "vaapi", StringComparison.OrdinalIgnoreCase)
                 && string.Equals(outputVideoCodec, "libx264", StringComparison.OrdinalIgnoreCase))
             {
                 var codec = videoStream.Codec.ToLowerInvariant();
-                var pixelFormat = videoStream.PixelFormat.ToLowerInvariant();
-
-                // Assert hardware VAAPI decodable (Except h264 10-bit and higher color depth)
-				// TODO: a propery way to detect hardware capabilities and falling back when transcoding is failed
-                if ((pixelFormat ?? string.Empty).IndexOf("p10", StringComparison.OrdinalIgnoreCase) == -1
-                    || ((pixelFormat ?? string.Empty).IndexOf("p10", StringComparison.OrdinalIgnoreCase) != -1
-                        && (string.Equals(codec, "hevc", StringComparison.OrdinalIgnoreCase)
-                            || string.Equals(codec, "h265", StringComparison.OrdinalIgnoreCase)
-                            || string.Equals(codec, "vp9", StringComparison.OrdinalIgnoreCase))))
+                var IsColorDepth10 = (videoStream.Profile ?? string.Empty).IndexOf("Main 10", StringComparison.OrdinalIgnoreCase) != -1
```

Comment:
```suggestion
                var isColorDepth10 = (videoStream.Profile ?? string.Empty).IndexOf("Main 10", StringComparison.OrdinalIgnoreCase) != -1
```
Since we are using netstandard2.1 now, we have access to the `string.Contains` method with a `StringComparison` overload.
Lastly, I would first check if the string isn't empty or null before the `Contains`